pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractTopic.java

```diff
@@ -205,6 +217,33 @@ protected void updateTopicPolicyByNamespacePolicy(Policies namespacePolicies) {
         Arrays.stream(BacklogQuota.BacklogQuotaType.values()).forEach(
                 type -> this.topicPolicies.getBackLogQuotaMap().get(type)
                         .updateNamespaceValue(MapUtils.getObject(namespacePolicies.backlog_quota_map, type)));
+        updateSchemaCompatibilityStrategyNamespaceValue(namespacePolicies);
+    }
+
+    private void updateSchemaCompatibilityStrategyNamespaceValue(Policies namespacePolicies){
+        if (this.isSystemTopic()) {
+            return;
+        }
+
+        topicPolicies.getSchemaCompatibilityStrategy().updateNamespaceValue(
+                formatSchemaCompatibilityStrategy(namespacePolicies.schema_compatibility_strategy));
+
+        // Backward compatibility: set schema_auto_update_compatibility_strategy value as broker value,
+        // it is the lowest priority.
+        // When the broker value is null in topic policies, set schema_auto_update_compatibility_strategy
+        // as broker value.
+        // When the broker value from topic policies does not equal from Pulsar config, need to update the
+        // schema_auto_update_compatibility_strategy to broker value.
+        SchemaCompatibilityStrategy strategyFromCache = topicPolicies.getSchemaCompatibilityStrategy().getBrokerValue();
+        SchemaCompatibilityStrategy strategyFromPulsarConfig = formatSchemaCompatibilityStrategy(
+                brokerService.pulsar().getConfiguration().getSchemaCompatibilityStrategy());
+
+        if (strategyFromCache == null || strategyFromCache != strategyFromPulsarConfig) {
```

Comment:
> strategyFromCache != strategyFromPulsarConfig

Could you please explain more details about this situation?

Reply:
We have `NamespacePolicies#schema_auto_update_compatibility_strategy` and `NamespacePolicies#schema_compatibility_strategy`, the `NamespacePolicies#schema_auto_update_compatibility_strategy` has been deprecated and the lowest priority, but we need to backward compatibility, so set this value to broker level of `topicPolicies.getSchemaCompatibilityStrategy()` when the broker level is null, once this value changed we need to update this value to broker level.

Reply:
I get your point.
IMO, this logic is not easy to understand. Is it the same as` strategyFromPulsarConfig == null` ?
As `schema_auto_update_compatibility_strategy` only take effects when broker level value is not set.

Reply:
> As `schema_auto_update_compatibility_strategy` only take effects when broker level value is not set.

Yes, if you have any good idea, please tell me.

Reply:
> Yes, if you have any good idea, please tell me.

This line could change to `if (strategyFromPulsarConfig == null)` ?

Reply:
There have two cases: 
1. when the broker value is null from the cache(`opicPolicies.getSchemaCompatibilityStrategy()`), we will set `schema_auto_update_compatibility_strategy` as broker value.
2. when the broker value in `broker.conf` does not equal the broker value from cache(`opicPolicies.getSchemaCompatibilityStrategy()`), it represents the `schema_auto_update_compatibility_strategy` has been changed, so we need to update this.